app/src/main/java/com/example/cr554/newsapp/NewsViewHolder.java

```diff
@@ -37,6 +39,8 @@ public void bindNews(NewsArticle news){
     }
 
     public void onClick(View view){
-        //opens a webview of the article
+        NewsArticle newsArticle = MainActivity.newsArticles.get(getAdapterPosition());
+        Intent browserIntent = new Intent("android.intent.action.VIEW", Uri.parse(newsArticle.getUrl()));
```

Comment:
This works too, but I believe the "new" way to do it is something like:

```
      Intent webIntent = new Intent(Intent.ACTION_VIEW)
          .setData(Uri.parse(getResources().getString(R.string.yale_youtube)))
          .addCategory(Intent.CATEGORY_BROWSABLE);
      startActivity(webIntent);
```

Notice few things:
1) You don't explicitly specify the string "android.intent.action.VIEW", which allows for better back-compatibility is Android decides to change the path (and that sometimes happens...)
2) You should likely move the URL to assets (better to use as few static public class members as possible)
3) This will use your browser to navigate the website. If you would like to use a webview you should do the following:

```
WebView dirWebView = (WebView) findViewById(R.id.directory_web_view);
    dirWebView.setWebViewClient(new WebViewClient());
    dirWebView.getSettings().setJavaScriptEnabled(true);
    dirWebView.loadUrl(getString(R.string.directory_address));
```

This of course assumes that I have a webview of the given ID and that R.string.directory_address contains the appropriate URL :-). This is a relatively new thing -- the webview will be completely ignored and a new browser activity launched if there is no WebViewClient and no ChromeWebViewClient :-)